carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/application/StudyService.kt

```diff
@@ -13,17 +14,37 @@ import dk.cachet.carp.studies.domain.users.Participant
 /**
  * Application service which allows creating and managing studies.
  */
+@Suppress( "TooManyFunctions" ) // TODO: Perhaps split up participation management from main interface.
```

Comment:
We should not do this to satisfy detekt defaults, but rather because it could make it a bit clearer what belongs where and that you are operating on different entities altogether

Reply:
We consider `Participant` a separate aggregate root (AR), which is only referred to from `Study` by ID through the list of participations. So, logically, it would make sense to split of management of distinct separate ARs into separate interfaces. I still think it is still useful to keep one 'study' endpoint, though, so think I can apply these changes without them having to impact you. It would simply involve interface delegation.